In [2]:
91 * 99

9009

A palindromic number reads the same both ways. The largest palindrome made from the product of two 2-digit numbers is 9009 = 91 × 99.

Find the largest palindrome made from the product of two 3-digit numbers.

In [50]:
from itertools import combinations,permutations,product
def calc_number(x,y,z,C):
    return(C*100*x + C*10*y + C*z)

check_palin = lambda x: x == int(str(x)[::-1])
palins_lit = []


for C in range(100,1000):
    print(C)
    for x,y,z in product(range(10),repeat=3):
        result = calc_number(x,y,z,C)
        if check_palin(result):
            print(result)
            print("x: {}, y: {}, z: {}".format(x,y,z))
            if(x != 0):
                palins_lit.append(result)
    print("---------------------------------------")



       

100
0
x: 0, y: 0, z: 0
---------------------------------------
101
0
x: 0, y: 0, z: 0
101
x: 0, y: 0, z: 1
202
x: 0, y: 0, z: 2
303
x: 0, y: 0, z: 3
404
x: 0, y: 0, z: 4
505
x: 0, y: 0, z: 5
606
x: 0, y: 0, z: 6
707
x: 0, y: 0, z: 7
808
x: 0, y: 0, z: 8
909
x: 0, y: 0, z: 9
1111
x: 0, y: 1, z: 1
2222
x: 0, y: 2, z: 2
3333
x: 0, y: 3, z: 3
4444
x: 0, y: 4, z: 4
5555
x: 0, y: 5, z: 5
6666
x: 0, y: 6, z: 6
7777
x: 0, y: 7, z: 7
8888
x: 0, y: 8, z: 8
9999
x: 0, y: 9, z: 9
10201
x: 1, y: 0, z: 1
11211
x: 1, y: 1, z: 1
12221
x: 1, y: 2, z: 1
13231
x: 1, y: 3, z: 1
14241
x: 1, y: 4, z: 1
15251
x: 1, y: 5, z: 1
16261
x: 1, y: 6, z: 1
17271
x: 1, y: 7, z: 1
18281
x: 1, y: 8, z: 1
19291
x: 1, y: 9, z: 1
20402
x: 2, y: 0, z: 2
21412
x: 2, y: 1, z: 2
22422
x: 2, y: 2, z: 2
23432
x: 2, y: 3, z: 2
24442
x: 2, y: 4, z: 2
25452
x: 2, y: 5, z: 2
26462
x: 2, y: 6, z: 2
27472
x: 2, y: 7, z: 2
28482
x: 2, y: 8, z: 2
29492
x: 2, y: 9, z: 2
30603
x: 3, y: 0, z: 3
31613
x: 3, y: 1, z: 3
32623
x: 3, y: 2, z: 

67876
x: 4, y: 7, z: 8
68586
x: 4, y: 8, z: 3
69296
x: 4, y: 8, z: 8
82928
x: 5, y: 8, z: 4
83638
x: 5, y: 8, z: 9
84348
x: 5, y: 9, z: 4
85058
x: 5, y: 9, z: 9
---------------------------------------
143
0
x: 0, y: 0, z: 0
858
x: 0, y: 0, z: 6
1001
x: 0, y: 0, z: 7
2002
x: 0, y: 1, z: 4
3003
x: 0, y: 2, z: 1
4004
x: 0, y: 2, z: 8
5005
x: 0, y: 3, z: 5
6006
x: 0, y: 4, z: 2
7007
x: 0, y: 4, z: 9
8008
x: 0, y: 5, z: 6
9009
x: 0, y: 6, z: 3
11011
x: 0, y: 7, z: 7
19591
x: 1, y: 3, z: 7
22022
x: 1, y: 5, z: 4
33033
x: 2, y: 3, z: 1
44044
x: 3, y: 0, z: 8
55055
x: 3, y: 8, z: 5
66066
x: 4, y: 6, z: 2
77077
x: 5, y: 3, z: 9
88088
x: 6, y: 1, z: 6
99099
x: 6, y: 9, z: 3
101101
x: 7, y: 0, z: 7
111111
x: 7, y: 7, z: 7
121121
x: 8, y: 4, z: 7
131131
x: 9, y: 1, z: 7
141141
x: 9, y: 8, z: 7
---------------------------------------
144
0
x: 0, y: 0, z: 0
6336
x: 0, y: 4, z: 4
21312
x: 1, y: 4, z: 8
27072
x: 1, y: 8, z: 8
42624
x: 2, y: 9, z: 6
48384
x: 3, y: 3, z: 6
63936
x: 4, y: 4, z: 4
69696
x

19691
x: 0, y: 9, z: 7
21112
x: 1, y: 0, z: 4
42224
x: 2, y: 0, z: 8
63336
x: 3, y: 1, z: 2
84448
x: 4, y: 1, z: 6
---------------------------------------
204
0
x: 0, y: 0, z: 0
21012
x: 1, y: 0, z: 3
42024
x: 2, y: 0, z: 6
63036
x: 3, y: 0, z: 9
84048
x: 4, y: 1, z: 2
---------------------------------------
205
0
x: 0, y: 0, z: 0
55555
x: 2, y: 7, z: 1
---------------------------------------
206
0
x: 0, y: 0, z: 0
21012
x: 1, y: 0, z: 2
26162
x: 1, y: 2, z: 7
42024
x: 2, y: 0, z: 4
47174
x: 2, y: 2, z: 9
63036
x: 3, y: 0, z: 6
68186
x: 3, y: 3, z: 1
84048
x: 4, y: 0, z: 8
89198
x: 4, y: 3, z: 3
---------------------------------------
207
0
x: 0, y: 0, z: 0
414
x: 0, y: 0, z: 2
828
x: 0, y: 0, z: 4
4554
x: 0, y: 2, z: 2
41814
x: 2, y: 0, z: 2
45954
x: 2, y: 2, z: 2
92529
x: 4, y: 4, z: 7
96669
x: 4, y: 6, z: 7
---------------------------------------
208
0
x: 0, y: 0, z: 0
27872
x: 1, y: 3, z: 4
42224
x: 2, y: 0, z: 3
84448
x: 4, y: 0, z: 6
---------------------------------------
209
0


239932
x: 9, y: 0, z: 2
---------------------------------------
267
0
x: 0, y: 0, z: 0
19491
x: 0, y: 7, z: 3
34443
x: 1, y: 2, z: 9
68886
x: 2, y: 5, z: 8
83838
x: 3, y: 1, z: 4
---------------------------------------
268
0
x: 0, y: 0, z: 0
27872
x: 1, y: 0, z: 4
46364
x: 1, y: 7, z: 3
---------------------------------------
269
0
x: 0, y: 0, z: 0
26362
x: 0, y: 9, z: 8
73437
x: 2, y: 7, z: 3
99799
x: 3, y: 7, z: 1
---------------------------------------
270
0
x: 0, y: 0, z: 0
---------------------------------------
271
0
x: 0, y: 0, z: 0
11111
x: 0, y: 4, z: 1
22222
x: 0, y: 8, z: 2
33333
x: 1, y: 2, z: 3
44444
x: 1, y: 6, z: 4
55555
x: 2, y: 0, z: 5
66666
x: 2, y: 4, z: 6
77777
x: 2, y: 8, z: 7
88888
x: 3, y: 2, z: 8
99999
x: 3, y: 6, z: 9
122221
x: 4, y: 5, z: 1
244442
x: 9, y: 0, z: 2
---------------------------------------
272
0
x: 0, y: 0, z: 0
272
x: 0, y: 0, z: 1
2992
x: 0, y: 1, z: 1
27472
x: 1, y: 0, z: 1
84048
x: 3, y: 0, z: 9
86768
x: 3, y: 1, z: 9
------------------------

66466
x: 1, y: 9, z: 9
---------------------------------------
335
0
x: 0, y: 0, z: 0
53935
x: 1, y: 6, z: 1
59295
x: 1, y: 7, z: 7
---------------------------------------
336
0
x: 0, y: 0, z: 0
48384
x: 1, y: 4, z: 4
65856
x: 1, y: 9, z: 6
---------------------------------------
337
0
x: 0, y: 0, z: 0
28982
x: 0, y: 8, z: 6
41114
x: 1, y: 2, z: 2
82228
x: 2, y: 4, z: 4
329923
x: 9, y: 7, z: 9
---------------------------------------
338
0
x: 0, y: 0, z: 0
676
x: 0, y: 0, z: 2
61516
x: 1, y: 8, z: 2
---------------------------------------
339
0
x: 0, y: 0, z: 0
49494
x: 1, y: 4, z: 6
78987
x: 2, y: 3, z: 3
95259
x: 2, y: 8, z: 1
324423
x: 9, y: 5, z: 7
---------------------------------------
340
0
x: 0, y: 0, z: 0
---------------------------------------
341
0
x: 0, y: 0, z: 0
4774
x: 0, y: 1, z: 4
5115
x: 0, y: 1, z: 5
9889
x: 0, y: 2, z: 9
31713
x: 0, y: 9, z: 3
56265
x: 1, y: 6, z: 5
87978
x: 2, y: 5, z: 8
129921
x: 3, y: 8, z: 1
133331
x: 3, y: 9, z: 1
184481
x: 5, y: 4, z: 1
215512


---------------------------------------
416
0
x: 0, y: 0, z: 0
27872
x: 0, y: 6, z: 7
84448
x: 2, y: 0, z: 3
---------------------------------------
417
0
x: 0, y: 0, z: 0
52125
x: 1, y: 2, z: 5
61716
x: 1, y: 4, z: 8
105501
x: 2, y: 5, z: 3
---------------------------------------
418
0
x: 0, y: 0, z: 0
8778
x: 0, y: 2, z: 1
63536
x: 1, y: 5, z: 2
204402
x: 4, y: 8, z: 9
221122
x: 5, y: 2, z: 9
239932
x: 5, y: 7, z: 4
256652
x: 6, y: 1, z: 4
273372
x: 6, y: 5, z: 4
290092
x: 6, y: 9, z: 4
408804
x: 9, y: 7, z: 8
---------------------------------------
419
0
x: 0, y: 0, z: 0
838
x: 0, y: 0, z: 2
56565
x: 1, y: 3, z: 5
308803
x: 7, y: 3, z: 7
---------------------------------------
420
0
x: 0, y: 0, z: 0
---------------------------------------
421
0
x: 0, y: 0, z: 0
34943
x: 0, y: 8, z: 3
52625
x: 1, y: 2, z: 5
70307
x: 1, y: 6, z: 7
---------------------------------------
422
0
x: 0, y: 0, z: 0
23632
x: 0, y: 5, z: 6
45154
x: 1, y: 0, z: 7
68786
x: 1, y: 6, z: 3
259952
x: 6, y: 1, z: 6


5335
x: 0, y: 1, z: 1
53835
x: 1, y: 1, z: 1
58685
x: 1, y: 2, z: 1
---------------------------------------
486
0
x: 0, y: 0, z: 0
---------------------------------------
487
0
x: 0, y: 0, z: 0
27272
x: 0, y: 5, z: 6
73537
x: 1, y: 5, z: 1
299992
x: 6, y: 1, z: 6
---------------------------------------
488
0
x: 0, y: 0, z: 0
40504
x: 0, y: 8, z: 3
69296
x: 1, y: 4, z: 2
445544
x: 9, y: 1, z: 3
---------------------------------------
489
0
x: 0, y: 0, z: 0
98289
x: 2, y: 0, z: 1
102201
x: 2, y: 0, z: 9
204402
x: 4, y: 1, z: 8
306603
x: 6, y: 2, z: 7
408804
x: 8, y: 3, z: 6
---------------------------------------
490
0
x: 0, y: 0, z: 0
---------------------------------------
491
0
x: 0, y: 0, z: 0
56465
x: 1, y: 1, z: 5
---------------------------------------
492
0
x: 0, y: 0, z: 0
---------------------------------------
493
0
x: 0, y: 0, z: 0
78387
x: 1, y: 5, z: 9
86768
x: 1, y: 7, z: 6
---------------------------------------
494
0
x: 0, y: 0, z: 0
494
x: 0, y: 0, z: 1
49894
x: 1, y: 0

x: 7, y: 1, z: 3
456654
x: 8, y: 1, z: 4
462264
x: 8, y: 2, z: 4
507705
x: 9, y: 0, z: 5
513315
x: 9, y: 1, z: 5
---------------------------------------
562
0
x: 0, y: 0, z: 0
25852
x: 0, y: 4, z: 6
83738
x: 1, y: 4, z: 9
---------------------------------------
563
0
x: 0, y: 0, z: 0
34343
x: 0, y: 6, z: 1
68686
x: 1, y: 2, z: 2
72627
x: 1, y: 2, z: 9
377773
x: 6, y: 7, z: 1
421124
x: 7, y: 4, z: 8
---------------------------------------
564
0
x: 0, y: 0, z: 0
27072
x: 0, y: 4, z: 8
29892
x: 0, y: 5, z: 3
297792
x: 5, y: 2, z: 8
---------------------------------------
565
0
x: 0, y: 0, z: 0
565
x: 0, y: 0, z: 1
51415
x: 0, y: 9, z: 1
---------------------------------------
566
0
x: 0, y: 0, z: 0
6226
x: 0, y: 1, z: 1
62826
x: 1, y: 1, z: 1
65656
x: 1, y: 1, z: 6
68486
x: 1, y: 2, z: 1
---------------------------------------
567
0
x: 0, y: 0, z: 0
---------------------------------------
568
0
x: 0, y: 0, z: 0
69296
x: 1, y: 2, z: 2
82928
x: 1, y: 4, z: 6
--------------------------------

---------------------------------------
638
0
x: 0, y: 0, z: 0
2552
x: 0, y: 0, z: 4
86768
x: 1, y: 3, z: 6
209902
x: 3, y: 2, z: 9
232232
x: 3, y: 6, z: 4
257752
x: 4, y: 0, z: 4
280082
x: 4, y: 3, z: 9
416614
x: 6, y: 5, z: 3
464464
x: 7, y: 2, z: 8
489984
x: 7, y: 6, z: 8
623326
x: 9, y: 7, z: 7
---------------------------------------
639
0
x: 0, y: 0, z: 0
84348
x: 1, y: 3, z: 2
---------------------------------------
640
0
x: 0, y: 0, z: 0
---------------------------------------
641
0
x: 0, y: 0, z: 0
528825
x: 8, y: 2, z: 5
---------------------------------------
642
0
x: 0, y: 0, z: 0
---------------------------------------
643
0
x: 0, y: 0, z: 0
81018
x: 1, y: 2, z: 6
---------------------------------------
644
0
x: 0, y: 0, z: 0
---------------------------------------
645
0
x: 0, y: 0, z: 0
53535
x: 0, y: 8, z: 3
588885
x: 9, y: 1, z: 3
---------------------------------------
646
0
x: 0, y: 0, z: 0
646
x: 0, y: 0, z: 1
611116
x: 9, y: 4, z: 6
----------------------------------

x: 7, y: 9, z: 1
575575
x: 8, y: 0, z: 5
585585
x: 8, y: 1, z: 9
595595
x: 8, y: 3, z: 3
---------------------------------------
716
0
x: 0, y: 0, z: 0
65156
x: 0, y: 9, z: 1
80908
x: 1, y: 1, z: 3
---------------------------------------
717
0
x: 0, y: 0, z: 0
717
x: 0, y: 0, z: 1
7887
x: 0, y: 1, z: 1
---------------------------------------
718
0
x: 0, y: 0, z: 0
---------------------------------------
719
0
x: 0, y: 0, z: 0
---------------------------------------
720
0
x: 0, y: 0, z: 0
---------------------------------------
721
0
x: 0, y: 0, z: 0
515515
x: 7, y: 1, z: 5
---------------------------------------
722
0
x: 0, y: 0, z: 0
63536
x: 0, y: 8, z: 8
698896
x: 9, y: 6, z: 8
---------------------------------------
723
0
x: 0, y: 0, z: 0
---------------------------------------
724
0
x: 0, y: 0, z: 0
---------------------------------------
725
0
x: 0, y: 0, z: 0
52925
x: 0, y: 7, z: 3
57275
x: 0, y: 7, z: 9
---------------------------------------
726
0
x: 0, y: 0, z: 0
23232
x: 0, 

---------------------------------------
805
0
x: 0, y: 0, z: 0
52325
x: 0, y: 6, z: 5
53935
x: 0, y: 6, z: 7
575575
x: 7, y: 1, z: 5
---------------------------------------
806
0
x: 0, y: 0, z: 0
434434
x: 5, y: 3, z: 9
---------------------------------------
807
0
x: 0, y: 0, z: 0
73437
x: 0, y: 9, z: 1
---------------------------------------
808
0
x: 0, y: 0, z: 0
808
x: 0, y: 0, z: 1
8888
x: 0, y: 1, z: 1
23432
x: 0, y: 2, z: 9
27472
x: 0, y: 3, z: 4
42824
x: 0, y: 5, z: 3
46864
x: 0, y: 5, z: 8
213312
x: 2, y: 6, z: 4
257752
x: 3, y: 1, z: 9
426624
x: 5, y: 2, z: 8
639936
x: 7, y: 9, z: 2
---------------------------------------
809
0
x: 0, y: 0, z: 0
---------------------------------------
810
0
x: 0, y: 0, z: 0
---------------------------------------
811
0
x: 0, y: 0, z: 0
25952
x: 0, y: 3, z: 2
---------------------------------------
812
0
x: 0, y: 0, z: 0
21112
x: 0, y: 2, z: 6
42224
x: 0, y: 5, z: 2
63336
x: 0, y: 7, z: 8
84448
x: 1, y: 0, z: 4
232232
x: 2, y: 8, z: 6
464464
x:

892
0
x: 0, y: 0, z: 0
461164
x: 5, y: 1, z: 7
---------------------------------------
893
0
x: 0, y: 0, z: 0
---------------------------------------
894
0
x: 0, y: 0, z: 0
855558
x: 9, y: 5, z: 7
---------------------------------------
895
0
x: 0, y: 0, z: 0
51015
x: 0, y: 5, z: 7
561165
x: 6, y: 2, z: 7
---------------------------------------
896
0
x: 0, y: 0, z: 0
48384
x: 0, y: 5, z: 4
---------------------------------------
897
0
x: 0, y: 0, z: 0
414414
x: 4, y: 6, z: 2
828828
x: 9, y: 2, z: 4
---------------------------------------
898
0
x: 0, y: 0, z: 0
898
x: 0, y: 0, z: 1
81718
x: 0, y: 9, z: 1
---------------------------------------
899
0
x: 0, y: 0, z: 0
9889
x: 0, y: 1, z: 1
---------------------------------------
900
0
x: 0, y: 0, z: 0
---------------------------------------
901
0
x: 0, y: 0, z: 0
78387
x: 0, y: 8, z: 7
---------------------------------------
902
0
x: 0, y: 0, z: 0
8118
x: 0, y: 0, z: 9
89298
x: 0, y: 9, z: 9
239932
x: 2, y: 6, z: 6
244442
x: 2, y: 7, z: 1

---------------------------------------
974
0
x: 0, y: 0, z: 0
27272
x: 0, y: 2, z: 8
299992
x: 3, y: 0, z: 8
---------------------------------------
975
0
x: 0, y: 0, z: 0
525525
x: 5, y: 3, z: 9
---------------------------------------
976
0
x: 0, y: 0, z: 0
69296
x: 0, y: 7, z: 1
---------------------------------------
977
0
x: 0, y: 0, z: 0
53735
x: 0, y: 5, z: 5
623326
x: 6, y: 3, z: 8
---------------------------------------
978
0
x: 0, y: 0, z: 0
204402
x: 2, y: 0, z: 9
408804
x: 4, y: 1, z: 8
---------------------------------------
979
0
x: 0, y: 0, z: 0
979
x: 0, y: 0, z: 1
329923
x: 3, y: 3, z: 7
378873
x: 3, y: 8, z: 7
436634
x: 4, y: 4, z: 6
485584
x: 4, y: 9, z: 6
543345
x: 5, y: 5, z: 5
592295
x: 6, y: 0, z: 5
601106
x: 6, y: 1, z: 4
650056
x: 6, y: 6, z: 4
---------------------------------------
980
0
x: 0, y: 0, z: 0
---------------------------------------
981
0
x: 0, y: 0, z: 0
64746
x: 0, y: 6, z: 6
---------------------------------------
982
0
x: 0, y: 0, z: 0
853358
x

In [51]:
max(palins_lit)

906609

In [43]:
list(range(100,1000))

[100,
 101,
 102,
 103,
 104,
 105,
 106,
 107,
 108,
 109,
 110,
 111,
 112,
 113,
 114,
 115,
 116,
 117,
 118,
 119,
 120,
 121,
 122,
 123,
 124,
 125,
 126,
 127,
 128,
 129,
 130,
 131,
 132,
 133,
 134,
 135,
 136,
 137,
 138,
 139,
 140,
 141,
 142,
 143,
 144,
 145,
 146,
 147,
 148,
 149,
 150,
 151,
 152,
 153,
 154,
 155,
 156,
 157,
 158,
 159,
 160,
 161,
 162,
 163,
 164,
 165,
 166,
 167,
 168,
 169,
 170,
 171,
 172,
 173,
 174,
 175,
 176,
 177,
 178,
 179,
 180,
 181,
 182,
 183,
 184,
 185,
 186,
 187,
 188,
 189,
 190,
 191,
 192,
 193,
 194,
 195,
 196,
 197,
 198,
 199,
 200,
 201,
 202,
 203,
 204,
 205,
 206,
 207,
 208,
 209,
 210,
 211,
 212,
 213,
 214,
 215,
 216,
 217,
 218,
 219,
 220,
 221,
 222,
 223,
 224,
 225,
 226,
 227,
 228,
 229,
 230,
 231,
 232,
 233,
 234,
 235,
 236,
 237,
 238,
 239,
 240,
 241,
 242,
 243,
 244,
 245,
 246,
 247,
 248,
 249,
 250,
 251,
 252,
 253,
 254,
 255,
 256,
 257,
 258,
 259,
 260,
 261,
 262,
 263,
 264,
 265,
 266